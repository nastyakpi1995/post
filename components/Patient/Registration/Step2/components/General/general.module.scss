.validation-general {
  padding-top: 40px;

  .validation-description {
    font-family: Avenir Next;
    font-style: normal;
    font-weight: 600;
    font-size: 16px;
    line-height: 20px;
    color: #252f4d;
    margin: 0;
  }
  .general-name {
    margin-top: 20px;

    .helper {
      font-family: Avenir Next;
      font-style: normal;
      font-weight: 500;
      font-size: 10px;
      letter-spacing: 1px;
      text-transform: uppercase;
      color: #bbbecd;
    }
  }
  .general-surname {
    margin-top: 20px;

    .helper {
      font-family: Avenir Next;
      font-style: normal;
      font-weight: 500;
      font-size: 10px;
      letter-spacing: 1px;
      text-transform: uppercase;
      color: #bbbecd;
    }
  }
  .general-id {
    margin-top: 30px;

    .helper {
      font-family: Avenir Next;
      font-style: normal;
      font-weight: 500;
      font-size: 10px;
      letter-spacing: 1px;
      text-transform: uppercase;
      color: #bbbecd;
    }
  }
}
