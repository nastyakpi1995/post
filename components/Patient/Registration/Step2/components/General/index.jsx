import React from 'react';
import Input from '../../../../../view/ui/Input';
import styles from './style.module.scss';

export default function General() {
  return (
    <div className={styles['validation-general']}>
      <p className={styles['validation-description']}>General information</p>
      <div className={styles['general-name']}>
        <label className={styles.helper}>
          first name
          <Input type="text" placeholder="Frederick" id="firstName" />
        </label>
      </div>
      <div className={styles['general-surname']}>
        <label className={styles.helper}>
          last name
          <Input type="text" placeholder="Valdez" id="lastName" />
        </label>
      </div>
      <div className={styles['general-id']}>
        <label className={styles.helper}>
          national id
          <Input type="number" placeholder="0123456789" id="nationalId" />
        </label>
      </div>
    </div>
  );
}
