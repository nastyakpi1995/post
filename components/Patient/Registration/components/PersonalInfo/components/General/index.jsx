import React from 'react';
import types from 'prop-types';
import { withTranslation } from 'i18n';
import { Input, FieldLabel, ErrorMessage } from '../../../../../../view/ui';
import styles from './style.module.scss';

function General({
  values,
  errors,
  touched,
  handleChange,
  handleBlur,
  t,
  personalInfoErrors,
}) {
  return (
    <div className={styles['validation-general']}>
      <p className={styles['validation-description']}>
        {t('personalInfo.general.title')}
      </p>
      <div className={styles['general-name']}>
        <FieldLabel text={t('personalInfo.general.firstName')}>
          <Input
            type="text"
            name="firstName"
            placeholder="John"
            value={values.firstName}
            onBlur={handleBlur}
            onChange={handleChange}
            error={
              errors.firstName && touched.firstName ? errors.firstName : null
            }
          />
          {errors.firstName && touched.firstName ? (
            <ErrorMessage text={errors.firstName} />
          ) : null}
        </FieldLabel>
      </div>
      <div className={styles['general-surname']}>
        <FieldLabel text={t('personalInfo.general.lastName')}>
          <Input
            type="text"
            name="lastName"
            placeholder="Frederick"
            value={values.lastName}
            onBlur={handleBlur}
            onChange={handleChange}
            error={errors.lastName && touched.lastName ? errors.lastName : null}
          />
          {errors.lastName && touched.lastName ? (
            <ErrorMessage text={errors.lastName} />
          ) : null}
        </FieldLabel>
      </div>
      <div className={styles['general-id']}>
        <FieldLabel text={t('personalInfo.general.nationalId')}>
          <Input
            type="text"
            name="nationalId"
            placeholder="12345678901"
            value={values.nationalId}
            onBlur={handleBlur}
            onChange={handleChange}
            error={
              errors.nationalId && touched.nationalId ? errors.nationalId : null
            }
          />
          {errors.nationalId && touched.nationalId ? (
            <ErrorMessage text={errors.nationalId} />
          ) : null}
          {personalInfoErrors
            ? personalInfoErrors.nationalId
              ? personalInfoErrors.nationalId.map((el) => {
                  return <ErrorMessage key={el} text={el} />;
                })
              : null
            : null}
        </FieldLabel>
      </div>
    </div>
  );
}

General.propTypes = {
  values: types.shape({
    firstName: types.string,
    lastName: types.string,
    nationalId: types.oneOfType([types.string, types.number]),
  }),
  errors: types.shape({
    firstName: types.string,
    lastName: types.string,
    nationalId: types.string,
  }),
  touched: types.shape({
    firstName: types.string,
    lastName: types.string,
    nationalId: types.string,
  }),
  handleChange: types.func,
  handleBlur: types.func,
  t: types.func,
};

export default withTranslation('registration')(General);
