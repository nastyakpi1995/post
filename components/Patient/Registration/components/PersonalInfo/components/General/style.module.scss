.validation-general {
  padding-top: 40px;

  .validation-description {
    font-family: Avenir Next;
    font-style: normal;
    font-weight: 600;
    font-size: 16px;
    line-height: 20px;
    color: #252f4d;
    margin: 0;
  }
  .general-name {
    margin-top: 20px;
  }
  .general-surname {
    margin-top: 20px;
  }
  .general-id {
    margin-top: 30px;

    & input::-webkit-outer-spin-button,
    & input::-webkit-inner-spin-button {
      -webkit-appearance: none;
      margin: 0;
    }

    & input[type=number] {
      -moz-appearance: textfield;
    }
  }
}
