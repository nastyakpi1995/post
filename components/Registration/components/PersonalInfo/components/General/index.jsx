import React from 'react';
import types from 'prop-types';
import { withTranslation } from 'i18n';
import { Input, FieldLabel, ErrorMessage } from '../../../../../view/ui';
import styles from './style.module.scss';

function General({
  values,
  errors,
  touched,
  handleChange,
  handleBlur,
  t,
  serverErrors = {},
  onResetErrors,
  type,
}) {
  return (
    <div className={styles['validation-general']}>
      <p className={styles['validation-description']}>
        {t('personalInfo.general.title')}
      </p>
      <div className={styles['general-name']}>
        <FieldLabel text={t('personalInfo.general.firstName')}>
          <Input
            type="text"
            name="firstName"
            placeholder="John"
            value={values.firstName}
            onBlur={handleBlur}
            onChange={handleChange}
            error={
              errors.firstName && touched.firstName ? errors.firstName : null
            }
          />
          {errors.firstName && touched.firstName ? (
            <ErrorMessage text={errors.firstName} />
          ) : null}
          {serverErrors.firstName
            ? serverErrors.firstName.map((el) => {
                return <ErrorMessage key={el} text={el} />;
              })
            : null}
        </FieldLabel>
      </div>
      <div className={styles['general-surname']}>
        <FieldLabel text={t('personalInfo.general.lastName')}>
          <Input
            type="text"
            name="lastName"
            placeholder="Frederick"
            value={values.lastName}
            onBlur={handleBlur}
            onChange={handleChange}
            error={errors.lastName && touched.lastName ? errors.lastName : null}
          />
          {errors.lastName && touched.lastName ? (
            <ErrorMessage text={errors.lastName} />
          ) : null}
          {serverErrors.lastName
            ? serverErrors.lastName.map((el) => {
                return <ErrorMessage key={el} text={el} />;
              })
            : null}
        </FieldLabel>
      </div>
      {type === 'DOC' ? (
        <div className={styles['doc-wrapper']}>
          <div className={styles['doc-inami']}>
            <FieldLabel text={t('personalInfo.general.enumNum')}>
              <Input type="text" onHandleChange={handleChange} />
            </FieldLabel>
          </div>
          <div className={styles['doc-qualification']}>
            <FieldLabel text={t('personalInfo.general.qualification')}>
              <Input type="text" onHandleChange={handleChange} />
            </FieldLabel>
          </div>
        </div>
      ) : null}
      <div className={styles['general-id']}>
        <FieldLabel text={t('personalInfo.general.nationalId')}>
          <Input
            type="text"
            name="nationalId"
            placeholder="12345678901"
            value={values.nationalId}
            onBlur={handleBlur}
            onChange={(e) => {
              if (serverErrors && serverErrors.nationalId) {
                onResetErrors({
                  nationalId: [],
                });
              }
              handleChange(e);
            }}
            error={
              errors.nationalId && touched.nationalId ? errors.nationalId : null
            }
          />
          {errors.nationalId && touched.nationalId ? (
            <ErrorMessage text={errors.nationalId} />
          ) : null}
          {serverErrors.nationalId
            ? serverErrors.nationalId.map((el) => {
                return <ErrorMessage key={el} text={el} />;
              })
            : null}
        </FieldLabel>
      </div>
    </div>
  );
}

General.propTypes = {
  values: types.shape({
    firstName: types.string,
    lastName: types.string,
    nationalId: types.oneOfType([types.string, types.number]),
  }),
  errors: types.shape({
    firstName: types.string,
    lastName: types.string,
    nationalId: types.string,
  }),
  touched: types.shape({
    firstName: types.string,
    lastName: types.string,
    nationalId: types.string,
  }),
  handleChange: types.func,
  handleBlur: types.func,
  t: types.func,
};

export default withTranslation('registration')(General);
