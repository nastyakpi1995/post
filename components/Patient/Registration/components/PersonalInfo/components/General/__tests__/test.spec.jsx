import React from 'react';
import renderer from 'react-test-renderer';
import General from '../index';

it('renders correctly', () => {
  const component = renderer.create(<General />).toJSON();
  expect(component).toMatchSnapshot();
});
