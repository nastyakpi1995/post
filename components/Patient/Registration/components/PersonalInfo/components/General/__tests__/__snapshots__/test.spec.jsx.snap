// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <p>
    General information
  </p>
  <div>
    <label>
      first name
      <div
        className="undefined"
      >
        <input
          id="firstName"
          name="firstName"
          placeholder="Frederick"
          type="text"
        />
      </div>
    </label>
  </div>
  <div>
    <label>
      last name
      <div
        className="undefined"
      >
        <input
          id="lastName"
          name="lastName"
          placeholder="Valdez"
          type="text"
        />
      </div>
    </label>
  </div>
  <div>
    <label>
      national id
      <div
        className="undefined"
      >
        <input
          id="nationalId"
          name="nationalId"
          placeholder="0123456789"
          type="number"
        />
      </div>
    </label>
  </div>
</div>
`;
